web/src/designer/components/Fields/BaseFieldEditor.tsx

```diff
@@ -170,15 +226,40 @@ export const BaseFieldEditor = ({fieldName, children}: Props) => {
       <Grid item xs={12}>
         <Card variant="outlined" sx={{p: 2}}>
           <Grid container spacing={2}>
-            <Grid item xs={12} md={6}>
+            <Grid item xs={12} md={4}>
               <DebouncedTextField
                 fullWidth
                 label="Label"
                 value={state.label}
                 onChange={e => updateProperty('label', e.target.value)}
               />
             </Grid>
-            <Grid item xs={12} md={6}>
+            <Grid item xs={12} md={4}>
+              <TextField
+                fullWidth
+                label="Field ID"
+                value={localFieldName}
+                onChange={handleIdChange}
+                inputRef={idInputRef}
+                InputProps={{
+                  endAdornment:
+                    state.label && slugify(state.label) !== localFieldName ? (
+                      <InputAdornment position="end">
+                        <Tooltip title="Sync with field name">
+                          <IconButton
+                            size="small"
+                            onClick={syncFieldID}
```

Comment:
There's an interaction with advanced helper text chechbox.  Click on that checkbox but don't enter any help text, then modify update the fieldID - the check box is unselected.